image_crop/recrop_min_size.py

```diff
@@ -294,18 +292,36 @@ def main():
                 skippedTiny.append((rowIndex, fileName, (maxX - minX) * (maxY - minY)))
                 continue
             # get base image from google drive that was uploaded by sort_images.py
-            dirID = getCameraDir(googleServices['drive'], cameraCache, fileName)
-            localFilePath = os.path.join(settings.downloadDir, fileName)
+            localFilePath = os.path.join(settings.downloadDir, fileName)#sets a path for that image() not yet downloadedby this iteration
             print('local', localFilePath)
-            if not os.path.isfile(localFilePath):
+            if not os.path.isfile(localFilePath):# if file has not been downloaded by a previous iteration
                 print('download', fileName)
-                goog_helper.downloadFile(googleServices['drive'], dirID, fileName, localFilePath)
-            imgOrig = Image.open(localFilePath)
+                nameParsed = img_archive.parseFilename(fileName)#parses file name into dictionary of parts name,unixtime,etc.
+                matchingCams = list(filter(lambda x: nameParsed['cameraID'] == x['id'], camArchives))#filter through camArchives for ids matching cameraid
+                if len(matchingCams) != 1:#if we cannot determine where the image will come from we cannot use the image
+                    logging.warning('Skipping camera without archive: %d, %s', len(matchingCams), str(matchingCams))
+                    skippedArchive.append((rowIndex, fileName, matchingCams))
+                    continue
+                archiveDirs = matchingCams[0]['dirs']
+                logging.warning('Found %s directories', archiveDirs)
+                time = datetime.datetime.fromtimestamp(nameParsed['unixTime'])
+                for dirName in archiveDirs:#search directories of camera for a time near
+                    logging.warning('Searching for files in dir %s', dirName)
+                    imgPaths = downloadFilesHttp(settings.downloadDir, nameParsed['cameraID'], dirName, time, time, 1, 0)
+                    if imgPaths:
+                        localFilePath = imgPaths[0]
+                        break
+                if not imgPaths:
+                    logging.warning('Skipping image not found: %s', fileName)
+                    skippedArchive.append((rowIndex, fileName, time))#archive that images were skipped
+                    continue
```

Comment:
You switched the code from using the full image uploaded to googleDrive by sort_images to downloading from HPWREN.  This has the benefit that we don't need to upload images in sort_images, but has the drawback of missing out images not in HPWREN, such as USGS.  If we want to do this, let's do it in a different change.